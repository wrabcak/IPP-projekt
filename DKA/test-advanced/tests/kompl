  

    # Konečný automat pro regulární výraz:
    # (a+c?|x*)*(b|c+d)+(((b*|d)a)*|x?y*z*|ax*y*)+
    # http://hackingoff.com/compilers/regular-expression-to-nfa-dfa
     
    (
        {
            s0, s1, s2, s3, s4, s5, s6, s7, s8, s9,
            s10, s11, s12, s13, s14, s15, s16, s17, s18, s19,
            s20, s21, s22, s23, s24, s25, s26, s27, s28, s29,
            s30, s31, s32, s33, s34, s35, s36, s37, s38, s39,
            s40, s41, s42, s43, s44, s45, s46, s47, s48, s49,
            s50, s51, s52, s53, s54
        },
     
        {
            'a', 'b', 'c', 'd',
            'x', 'y', 'z'
        },
     
        {
            s0 '' -> s1, s0 '' -> s3,
            s1 '' -> s4, s1 '' -> s6,
            s2 '' -> s3, s2 '' -> s1,
            s3 '' -> s13,
            s4 '' -> s8,
            s5 '' -> s2,
            s6 '' -> s7, s6 '' -> s11,
            s7 '' -> s2,
            s8 'a' -> s9,
            s9 '' -> s8, s9 '' -> s10, #15
     
            s10 'c' -> s5, s10 '' -> s5,
            s11 'x' -> s12,
            s12 '' -> s7, s12 '' -> s11,
            s13 '' -> s16, s13 '' -> s18,
            s14 '' -> s15, s14 '' -> s13,
            s15 '' -> s23,
            s16 'b' -> s17,
            s17 '' -> s14,
            s18 '' -> s20,
            s19 '' -> s14, #14
     
            s20 'c' -> s21,
            s21 '' -> s20, s21 '' -> s22,
            s22 'd' -> s19,
            s23 '' -> s26, s23 '' -> s28,
            s24 '' -> s25, s24 '' -> s23,
            #s25
            s26 '' -> s30, s26 '' -> s27,
            s27 '' -> s24,
            s28 '' -> s39, s28 '' -> s41,
            s29 '' -> s24, #13
     
            s30 '' -> s32, s30 '' -> s34,
            s31 '' -> s27, s31 '' -> s30,
            s32 '' -> s33, s32 '' -> s37,
            s33 '' -> s36,
            s34 'd' -> s35,
            s35 '' -> s36,
            s36 'a' -> s31,
            s37 'b' -> s38,
            s38 '' -> s37, s38 '' -> s33,
            s39 'x' -> s43, s39 '' -> s43, #15
     
            s40 '' -> s29,
            s41 'a' -> s49,
            s42 '' -> s29,
            s43 '' -> s44, s43 '' -> s46,
            s44 'y' -> s45,
            s45 '' -> s44, s45 '' -> s46,
            s46 '' -> s47, s46 '' -> s40,
            s47 'z' -> s48,
            s48 '' -> s40, s48 '' -> s47,
            s49 '' -> s50, s49 '' -> s52, #15
     
            s50 'x' -> s51,
            s51 '' -> s50, s51 '' -> s52,
            s52 '' -> s42, s52 '' -> s53,
            s53 'y' -> s54,
            s54 '' -> s53, s54 '' -> s42 #8
        }, # 8+15+15+13+14+15=80
     
        s0,
     
        {
            s25
        }
    )

