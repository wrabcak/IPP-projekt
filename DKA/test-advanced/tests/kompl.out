(
{s0, s12, s17, s17_s38, s19, s21, s21_s5, s31, s31_s49, s35, s38, s43, s43_s51, s45, s45_s54, s48, s9},
{'a', 'b', 'c', 'd', 'x', 'y', 'z'},
{
s0 'a' -> s9,
s0 'b' -> s17,
s0 'c' -> s21,
s0 'x' -> s12,
s12 'a' -> s9,
s12 'b' -> s17,
s12 'c' -> s21,
s12 'x' -> s12,
s17 'a' -> s31_s49,
s17 'b' -> s17_s38,
s17 'c' -> s21,
s17 'd' -> s35,
s17 'x' -> s43,
s17 'y' -> s45,
s17 'z' -> s48,
s17_s38 'a' -> s31_s49,
s17_s38 'b' -> s17_s38,
s17_s38 'c' -> s21,
s17_s38 'd' -> s35,
s17_s38 'x' -> s43,
s17_s38 'y' -> s45,
s17_s38 'z' -> s48,
s19 'a' -> s31_s49,
s19 'b' -> s17_s38,
s19 'c' -> s21,
s19 'd' -> s35,
s19 'x' -> s43,
s19 'y' -> s45,
s19 'z' -> s48,
s21 'c' -> s21,
s21 'd' -> s19,
s21_s5 'a' -> s9,
s21_s5 'b' -> s17,
s21_s5 'c' -> s21,
s21_s5 'd' -> s19,
s21_s5 'x' -> s12,
s31 'a' -> s31_s49,
s31 'b' -> s38,
s31 'd' -> s35,
s31 'x' -> s43,
s31 'y' -> s45,
s31 'z' -> s48,
s31_s49 'a' -> s31_s49,
s31_s49 'b' -> s38,
s31_s49 'd' -> s35,
s31_s49 'x' -> s43_s51,
s31_s49 'y' -> s45_s54,
s31_s49 'z' -> s48,
s35 'a' -> s31,
s38 'a' -> s31,
s38 'b' -> s38,
s43 'a' -> s31_s49,
s43 'b' -> s38,
s43 'd' -> s35,
s43 'x' -> s43,
s43 'y' -> s45,
s43 'z' -> s48,
s43_s51 'a' -> s31_s49,
s43_s51 'b' -> s38,
s43_s51 'd' -> s35,
s43_s51 'x' -> s43_s51,
s43_s51 'y' -> s45_s54,
s43_s51 'z' -> s48,
s45 'a' -> s31_s49,
s45 'b' -> s38,
s45 'd' -> s35,
s45 'x' -> s43,
s45 'y' -> s45,
s45 'z' -> s48,
s45_s54 'a' -> s31_s49,
s45_s54 'b' -> s38,
s45_s54 'd' -> s35,
s45_s54 'x' -> s43,
s45_s54 'y' -> s45_s54,
s45_s54 'z' -> s48,
s48 'a' -> s31_s49,
s48 'b' -> s38,
s48 'd' -> s35,
s48 'x' -> s43,
s48 'y' -> s45,
s48 'z' -> s48,
s9 'a' -> s9,
s9 'b' -> s17,
s9 'c' -> s21_s5,
s9 'x' -> s12
},
s0,
{s17, s17_s38, s19, s31, s31_s49, s43, s43_s51, s45, s45_s54, s48}
)